src/editors/textEditor/textEditor.js

```diff
@@ -445,6 +445,10 @@ export class TextEditor extends BaseEditor {
   bindEvents() {
     this.eventManager.addEventListener(this.TEXTAREA, 'cut', event => event.stopPropagation());
     this.eventManager.addEventListener(this.TEXTAREA, 'paste', event => event.stopPropagation());
+    if (isIOS) {
+      // on iOS after click "Done" the edit isn't hidden by default, so we need to handle it manually.
+      this.eventManager.addEventListener(this.TEXTAREA, 'focusout', () => this.editorFocusOut());
```

Comment:
```suggestion
      this.eventManager.addEventListener(this.TEXTAREA, 'focusout', () => this.finishEditing(false));
```
When the editor is only hidden the newly entered value is not visible after tapping the "done" button. As you can see here: 

https://user-images.githubusercontent.com/571316/114152318-6e16f900-991e-11eb-826c-d651b5f4495c.mov

It seems that it can be a better idea to use the `finishEditing` method (so the `editorFocusOut` could be removed). Thanks to that after hitting the "done" button the value is immediately visible in the table. 

https://user-images.githubusercontent.com/571316/114152283-63f4fa80-991e-11eb-9520-7b0c7b30fae7.mov

It's worth adding the test for that behavior (https://github.com/handsontable/handsontable/tree/feature/issue-7659/test/e2e/mobile).